### PR TITLE
Support .NET 5.0 pipe security.

### PR DESCRIPTION
**Pull request by ccruden-shookiot**

### CODE REVIEW

**Comment by ccruden-shookiot** [comment]
@googlebot I signed it!

**Comment by ccruden-shookiot** [comment]
Seems to be building and testing mostly properly at this point....
Failing CancelServerStreaming in the GitHub build, but the comment in code seems to indicate that test is flaky, and it doesn't fail when I run it locally...

File: GrpcDotNetNamedPipes/GrpcDotNetNamedPipes.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFrameworks>net46;netcoreapp2.1;netstandard2.1;netstandard2.0</TargetFrameworks>
+        <TargetFrameworks>net46;netcoreapp3.1;netstandard2.1;netstandard2.0;net5.0-windows</TargetFrameworks>
```
Comment (cyanfish) [inline GrpcDotNetNamedPipes/GrpcDotNetNamedPipes.csproj:4]: Same here.

File: GrpcDotNetNamedPipes.Tests/GrpcDotNetNamedPipes.Tests.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFrameworks>net46;netcoreapp2.1</TargetFrameworks>
+        <TargetFrameworks>net46;netcoreapp3.1;net5.0-windows</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (ccruden-shookiot) [reply GrpcDotNetNamedPipes.Tests/GrpcDotNetNamedPipes.Tests.csproj:4]: No, it doesn't. Forgot I'd changed that back from the original attempt to get .NET Core 3.1 working. I'll change that back.

**Comment by cyanfish** [comment]
Also ignore the test error around .NET core not being available, that's a preexisting issue that must've been caused by a change to the github CI environment.

**Pull request merged**